homeassistant/components/sabnzbd/__init__.py

```diff
@@ -1,234 +1,58 @@
-                vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
-                vol.Optional(CONF_PATH): cv.string,
-                vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-                vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-                vol.Optional(CONF_SENSORS): vol.All(
-                    cv.ensure_list, [vol.In(SENSOR_KEYS)]
-                ),
-                vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
-            }
-        )
-    },
-    extra=vol.ALLOW_EXTRA,
-)
-
 
-async def async_check_sabnzbd(sab_api):
-    """Check if we can reach SABnzbd."""
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up the SabNzbd Component."""
+    sab_api = await get_client(hass, entry.data)
 
-    try:
-        await sab_api.check_available()
-        return True
-    except SabnzbdApiException:
-        _LOGGER.error("Connection to SABnzbd API failed")
-        return False
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        KEY_API: sab_api,
+        KEY_NAME: entry.data[CONF_NAME],
+    }
 
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
 
-async def async_configure_sabnzbd(
-    hass, config, use_ssl, name=DEFAULT_NAME, api_key=None
-):
-    """Try to configure Sabnzbd and request api key if configuration fails."""
+    async_setup_sabnzbd(hass, sab_api)
 
-    host = config[CONF_HOST]
-    port = config[CONF_PORT]
-    web_root = config.get(CONF_PATH)
-    uri_scheme = "https" if use_ssl else "http"
-    base_url = BASE_URL_FORMAT.format(uri_scheme, host, port)
-    if api_key is None:
-        conf = await hass.async_add_executor_job(
-            load_json, hass.config.path(CONFIG_FILE)
-        )
-        api_key = conf.get(base_url, {}).get(CONF_API_KEY, "")
-
-    sab_api = SabnzbdApi(
-        base_url, api_key, web_root=web_root, session=async_get_clientsession(hass)
-    )
-    if await async_check_sabnzbd(sab_api):
-        async_setup_sabnzbd(hass, sab_api, config, name)
-    else:
-        async_request_configuration(hass, config, base_url, web_root)
-
-
-async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up the SABnzbd component."""
-
-    async def sabnzbd_discovered(service: str, info: DiscoveryInfoType | None) -> None:
-        """Handle service discovery."""
-        if not info:
-            return
-        ssl = info.get("properties", {}).get("https", "0") == "1"
-        await async_configure_sabnzbd(hass, info, ssl)
-
-    discovery.async_listen(hass, SERVICE_SABNZBD, sabnzbd_discovered)
-
-    if (conf := config.get(DOMAIN)) is not None:
-        use_ssl = conf[CONF_SSL]
-        name = conf.get(CONF_NAME)
-        api_key = conf.get(CONF_API_KEY)
-        await async_configure_sabnzbd(hass, conf, use_ssl, name, api_key)
     return True
 
 
 @callback
-def async_setup_sabnzbd(hass, sab_api, config, name):
+def async_setup_sabnzbd(hass, sab_api):
```

Comment:
Why a standalone method ?

Reply:
not sure what you mean

Reply:
it's only used once, why not make it part of async_setup_entry

Reply:
done